src/components/Card.js

```diff
@@ -1,37 +1,69 @@
 export default class Card {
-  constructor(data, cardSelector, handleImageClick) {
+  constructor(
+    data,
+    cardSelector,
+    handleImageClick,
+    handleLikeIcon,
+    handleDeleteCard
+  ) {
     this._name = data.name;
     this._link = data.link;
+    this.id = data._id;
+    this._isLiked = data.isLiked;
     this._cardSelector = cardSelector;
     this._handleImageClick = handleImageClick;
+    this._handleLikeIcon = handleLikeIcon;
+    this.handleDeleteCard = handleDeleteCard;
   }
 
   _setEventListeners() {
     this._cardElement
-      .querySelector(".card__like-button")
+      .querySelector(".card__delete-button")
       .addEventListener("click", () => {
-        this._handleLikeButton();
+        this.handleDeleteCard(this);
       });
     this._cardElement
-      .querySelector(".card__delete-button")
+      .querySelector(".card__image")
```

Comment:
You've already found `this._cardImageElement` in line 79. Please, use it everywhere in `Card`

`Class fields` (the so-called **class variables** with `this`) are accessible inside any class method. You shouldn’t pass **class fields** into the call of any method or search them again because you can use class fields inside all methods.

Reply:
Thanks for that tip! Does it make sense then to just set all class fields in the constructor, as opposed to defining new things within methods? Or should the constructor itself not be too bulky?

Reply:
> Or should the constructor itself not be too bulky

The class fields are usually defined in the method which is called **first** of all.  But the `cosntructor` is also a nice place for defining to make sure that any other method will have them